### PR TITLE
Od/popup banner

### PR DESCRIPTION
**Pull request by orlando**
**What:** Add popup banner component

image

https://www.loom.com/share/5308c290b72a4c1c935261c829de3edf

**Why:** To reuse this component in the landing page and other pages. I had to modify this component a bit in order to fit the message the product team provided. It uses a one-column layout.

**How:**

- Port code from strike website
- Build a component with stenciljs and publish it to npm. https://www.npmjs.com/package/@debtcollective/dc-popup-component

### CODE REVIEW

File: packages/popup-component/src/components.d.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+
+export namespace Components {
+  interface DcPopup {
+    /**
+    * Date
+    */
+    'date': string;
+    /**
+    * Hero
```
Comment (duranmla) [inline packages/popup-component/src/components.d.ts]: Just small comment "hero" doesn't seems to be a easy name we can related to "what it represents within the popup", I would rather use "title"

File: packages/popup-component/src/components/popup/popup.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { newE2EPage } from '@stencil/core/testing';
+
+describe('my-component', () => {
```
Comment (duranmla) [inline packages/popup-component/src/components/popup/popup.e2e.ts]: We don't need this test isn't?

File: packages/popup-component/src/components/popup/popup.scss [hunk 3]
```diff
@@ -0,0 +1,97 @@
+@import "shared.scss";
+
+.close {
+  color: #000;
```
Comment (duranmla) [inline packages/popup-component/src/components/popup/popup.scss]: If we have variables _(as I see below)_ use variables instead.

File: packages/popup-component/src/components/popup/popup.scss [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  cursor: pointer;
+  float: right;
+  font-size: 2.25rem;
+  font-weight: 600;
+  line-height: 1;
+  opacity: 0.5;
+  text-shadow: 0 1px 0 #fff;
```
Comment (duranmla) [inline packages/popup-component/src/components/popup/popup.scss]: Variables and rem units to keep everything may be missing here.

File: packages/popup-component/src/components/popup/popup.scss [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+  &:not(:disabled):not(.disabled):hover,
+  &:not(:disabled):not(.disabled):focus {
+    opacity: 0.75;
+  }
+}
+
+.notification {
+  color: $gray;
+  font-family: "Libre Franklin", sans-serif;
```
Comment (duranmla) [inline packages/popup-component/src/components/popup/popup.scss]: If we have bootstrap variable you probably have `$font-family-sans-serif`

File: packages/popup-component/src/components/popup/shared.scss [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+$purple: #dac4f5;
+$green: #d6ffb8;
+$yellow: #ffed9c;
+$gray: #2b2b2b;
+$beige: #faf9f5;
+$white: #ffffff;
+$black: #000000;
+$grid-gutter-width: 40px;
+$zindex-fixed: 1030;
+$xs: 576px;
```
Comment (duranmla) [inline packages/popup-component/src/components/popup/shared.scss]: You have `$grid-breakpoints` which you can use like: `map-get($map: $grid-breakpoints, $key: "xs")`

File: packages/popup-component/src/utils/utils.spec.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { format } from './utils';
+
+describe('format', () => {
```
Comment (duranmla) [inline packages/popup-component/src/utils/utils.spec.ts]: If we don't need this one we should remove it.

File: packages/popup-component/src/components.d.ts (hunk 1 above)
Comment (orlando) [reply packages/popup-component/src/components.d.ts]: title is reserved in Stenciljs, that’s why I changed it to hero. Open to suggestions on better naming

File: packages/popup-component/src/components/popup/popup.e2e.ts (hunk 2 above)
Comment (orlando) [reply packages/popup-component/src/components/popup/popup.e2e.ts]: No, but we can add a small render test using this file

File: packages/popup-component/src/components/popup/popup.scss (hunk 3 above)
Comment (orlando) [reply packages/popup-component/src/components/popup/popup.scss]: This was copied I think from the other file, also we have a lot of dependencies to bootstrap. I just copied these dependencies over for now but I would like to check a better way to do it, im open to suggestions

File: packages/popup-component/src/components/popup/popup.scss (hunk 5 above)
Comment (orlando) [reply packages/popup-component/src/components/popup/popup.scss]: We don’t have bootstrap or bootstrap variables here

File: packages/popup-component/src/components/popup/shared.scss (hunk 6 above)
Comment (orlando) [reply packages/popup-component/src/components/popup/shared.scss]: We don’t have bootstrap or bootstrap variables here, for these kind of things I don’t think we should really on bootstrap

**Pull request merged**